src/MicroEd-Tests/MDFileTest.class.st

```diff
@@ -0,0 +1,40 @@
+Class {
+	#name : 'MDFileTest',
+	#superclass : 'TestCase',
+	#instVars : [
+		'memoryFS',
+		'fileExample'
+	],
+	#category : 'MicroEd-Tests',
+	#package : 'MicroEd-Tests'
+}
+
+{ #category : 'running' }
+MDFileTest >> setUp [
+
+	super setUp.
+	memoryFS := FileSystem memory.
+	fileExample := memoryFS / 'example.txt'.
+	fileExample writeStreamDo: [ :stream |
+		stream nextPutAll: 'example file' ]
+]
+
+{ #category : 'running' }
+MDFileTest >> testMDFileWithFile [
+
+	| file |
+	file := MDFile new.
+	file fromFile: fileExample.
+	self assert: file hasFile.
+	self assert: file basename equals: fileExample basename.
+	self assert: file extension equals: fileExample extension
+]
+
+{ #category : 'running' }
+MDFileTest >> testMDFileWithoutFile [
```

Comment:
I do not get it. If you do not specify a file why does it have one?
To me this is not correct.

Reply:
This is when you "create" a new file object that is still unsaved by the user. 
For instance, see the FSM transition graph for a typical project here: https://github.com/cormas/ProjectFramework?tab=readme-ov-file#finite-state-machine. That could be applied to a file.